dev_guide/application_lifecycle.adoc

```diff
@@ -0,0 +1,77 @@
+= The Application Life Cycle
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+As a PaaS, OpenShift is built for deploying applications. Depending on how much
+you want to involve OpenShift in the development process, you can choose to
+develop on OpenShift after it has been installed, and use it to continuously
+develop an application, or you can deploy a fully developed application onto a
+newly installed OpenShift instance.
+
+== Developing on OpenShift
+
+If you plan to use develop your application on OpenShift directly:
+
+Plan::
+* What does your application do?
+* What programming language will you be developing in?
+ 
+Install::
+* OpenShift should be installed by this point, either by yourself or an
+administrator within your organization.
+
+Develop::
+* This is where you develop your application code.
```

Comment:
Why does "develop" come before "generate" and "code"?  It seems like "generate" and "code" are two sub-steps of "develop"